### ISSUE TITLE
Add support for different repos for different distributions

### ISSUE THREAD

**Issue by tfoote** [ros-infrastructure/ros_buildfarm#297]
Tasks which will be necessary to support deploying a new gpg key in parallel with the old one in a parallel repo. Needed for option 2 in this discussion: http://discourse.ros.org/t/deployment-approaches-for-new-gpg-key/224/2

ros_buildfarm:

* [ ] extend import_upstream to pull into a parameterized list of repos (currently hard coded 3 names)
* [ ] parameterize sync packages to main to take target repos as arguments
* [ ] parameterize sync packages_to_testing to take target repos as arguments.
* [ ] add support to buildfiles to store above repo sequence, if unset keep current defaults

ros_buildfarm_config: 

* [ ] add element to call out repo sequence to build files

buildfarm_deployment_config

* [ ] extend reprepro config elements to use the new paths and gpg keys.

## Tasks which can be deferred by doing manual maintenance

buildfarm_deployment

* support installing multiple gpg keys (workaround manually install the additional keys) https://github.com/ros-infrastructure/buildfarm_deployment/issues/128
* support initiallizing a parameterized list of repos. currently hardcoded (workaround manually iniit new repos) https://github.com/ros-infrastructure/buildfarm_deployment/issues/129

ros_buildfarm

* remove hardcoded repo path in import_package_job #298

**Comment by tfoote** [comment]
We found a way to do this on the ROS infrastructure w/o needing to fork the repos. https://github.com/ros-infrastructure/buildfarm_deployment/issues/130

This may be a feature we want in the future. But the other option is to simply have separate farms and keep the complexity down. I"m going to close as won't fix for now.
